Unhandled exception `key_has_unknown_prefix/1`

```shell
$ curl -X POST -u admin:root 'http://localhost:6363/api/document/_system?author=admin&message=test' \ 
   -d '{ "@type": "foo:User" }' -H 'Content-Type: application/json'
```

```
[ERROR] 2021-12-13T15:57:36.212902+01:00 [Exception] error(key_has_unknown_prefix("foo:User"),context(prolog_stack([frame(48,call(system:throw/1),throw(error(key_has_unknown_prefix("foo:User"),_14956))),frame(46,pred_line("catch/3",'/home/gavin/.swivm/versions/v8.4.1/lib/swipl/boot/init.pl':562),catch(api_document:do_or_die(...,...),error(key_has_unknown_prefix("system:User"),_15012)
```

Just needs to be caught and handled.

--- a/src/core/api/api_error.pl
+++ b/src/core/api/api_error.pl
@@ -1958,6 +1958,15 @@ api_document_error_jsonld(Type,error(not_a_unit_type(Value, Document), _), JSON)
                               'api:document': Document},
              'api:message' : Msg
             }.
+api_document_error_jsonld(Type,error(key_has_unknown_prefix(Prefixed_Key), _), JSON) :-
+    document_error_type(Type, Type_Displayed),
+    format(string(Msg), "The key '~w' is utilising an unknown prefix.", [Prefixed_Key]),
+    JSON = _{'@type' : Type_Displayed,
+             'api:status' : "api:failure",
+             'api:message' : Msg,
+             'api:error' : _{ '@type' : "api:UnknownPrefixError",
+                              'api:key' : Prefixed_Key}
+            }.
 
 /**
  * generic_exception_jsonld(Error,JSON) is det.

--- a/tests/test/document-auth.js
+++ b/tests/test/document-auth.js
@@ -514,6 +514,13 @@ describe('document', function () {
         const id = `terminusdb:///data/${instance['@id']}`
         await document.insert(agent, { instance, raw_json: true }).fails(api.error.invalidJSONDocumentId(id))
       })
+
+      it('fails gracefully with bad key prefix', async function () {
+        const instance = { '@type' : "foo:User" }
+        const result = await document.insert(agent, { instance }).unverified()
+        expect(result.status).to.equal(400)
+        expect(result.body['api:error']['api:key']).to.equal('foo:User')
+      })
     })
   })
 })

--- a/tests/test/document-auth.js
+++ b/tests/test/document-auth.js
@@ -516,7 +516,7 @@ describe('document', function () {
       })
 
       it('fails gracefully with bad key prefix', async function () {
-        const instance = { '@type' : "foo:User" }
+        const instance = { '@type': 'foo:User' }
         const result = await document.insert(agent, { instance }).unverified()
         expect(result.status).to.equal(400)
         expect(result.body['api:error']['api:key']).to.equal('foo:User')

